lib/transaction/priority_queue.go

```diff
@@ -134,11 +135,43 @@ func (spq *PriorityQueue) Push(txn *ValidTransaction) (common.Hash, error) {
 	spq.currOrder++
 	heap.Push(&spq.pq, item)
 	spq.txs[hash] = item
+	if spq.nextPushWatcher != nil {
+		select {
+		case _, ok := <-spq.nextPushWatcher:
+			if ok {
+				close(spq.nextPushWatcher)
+			}
+		default:
+		}
+	}
 
 	transactionQueueGauge.Set(float64(spq.pq.Len()))
 	return hash, nil
 }
 
+// NewPushWatcher returns a read only channel to be signalled
+// when the next Push() is called on the queue.
+// Note the returned channel is closed when the Push() function
+// is called.
+func (spq *PriorityQueue) NewPushWatcher() (nextPushWatcher <-chan struct{}) {
```

Comment:
nit `NextPushWatcher` would fit better, since we are not creating a watcher every time